internal/transport/http2_server.go

```diff
@@ -403,6 +403,10 @@ func (t *http2Server) operateHeaders(frame *http2.MetaHeadersFrame, handle func(
 			mdata[hf.Name] = append(mdata[hf.Name], hf.Value)
 			s.contentSubtype = contentSubtype
 			isGRPC = true
+
+		case "grpc-accept-encoding":
+			s.clientAdvertisedCompressors = hf.Value
+			mdata[hf.Name] = append(mdata[hf.Name], hf.Value)
```

Comment:
We were exposing this already?  It seems like it.  That's not really desirable, but I don't think we can change it, unfortunately.

Would you mind adding an accessor that returns this, and document using it to determine how to use `SetSendCompressor`, to avoid having people rely on it being in the raw metadata?

Reply:
Done, exposed a new method `ClientAdvertisedCompressors` alongside `SetSendCompressor` with references in method doc